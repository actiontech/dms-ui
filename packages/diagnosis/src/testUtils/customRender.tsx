import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter, MemoryRouterProps } from 'react-router-dom';
import { Dictionary } from '@actiontech/shared/lib/types/common.type';
import { storeFactory } from './mockRedux';
import { StyledEngineProvider, ThemeProvider } from '@mui/system';
import { mount } from 'enzyme';
import { ConfigProvider } from 'antd';
import zhCN from 'antd/locale/zh_CN';

import lightTheme from '../theme/light';
import sharedTheme from '@actiontech/shared/lib/theme/light';

type MountParams = Parameters<typeof mount>;
type RenderParams = Parameters<typeof render>;

export const renderWithReduxAndTheme = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(ui, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={{ ...lightTheme, ...sharedTheme }}>
        <Provider store={storeFactory(initStore)}>{children}</Provider>
      </ThemeProvider>
    ),
    ...Option
  });
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(
    <ThemeProvider theme={{ ...lightTheme, ...sharedTheme }}>
      {ui}
    </ThemeProvider>,
    option
  );
};

export const renderWithThemeAndRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(
    <MemoryRouter {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={{ ...lightTheme, ...sharedTheme }}>
          {ui}
        </ThemeProvider>
      </StyledEngineProvider>
    </MemoryRouter>,
    option
  );
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <ConfigProvider locale={zhCN}>
          <ThemeProvider theme={{ ...lightTheme, ...sharedTheme }}>
            <MemoryRouter {...otherProps?.routerProps}>
              <Provider store={storeFactory(otherProps?.initStore)}>
                {children}
              </Provider>
            </MemoryRouter>
          </ThemeProvider>
        </ConfigProvider>
      );
    },
    ...option
  });
  return renderReturn;
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: { ...lightTheme, ...sharedTheme }
    }
  });
};
