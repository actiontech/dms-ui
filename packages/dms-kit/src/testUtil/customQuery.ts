import * as domTestingLib from '@testing-library/dom';
import { act, fireEvent, screen } from '@testing-library/react';

export const getBySelector = (selector: string, baseElement?: Element) => {
  let temp;
  if (baseElement !== undefined) {
    temp = baseElement.querySelectorAll(selector);
  } else {
    temp = document.querySelectorAll(selector);
  }
  if (temp.length === 0) {
    throw domTestingLib.queryHelpers.getElementError(
      `Unable to find an element by: selector=${selector}`,
      document.body
    );
  }
  if (temp.length > 1) {
    throw domTestingLib.queryHelpers.getElementError(
      `Found multiple elements with the selector=${selector}`,
      document.body
    );
  }
  return temp[0];
};

export const queryBySelector = (selector: string, baseElement?: Element) => {
  let temp;
  if (baseElement !== undefined) {
    temp = baseElement.querySelector(selector);
  } else {
    temp = document.querySelector(selector);
  }
  return temp;
};

export const getAllBySelector = (selector: string, baseElement?: Element) => {
  let temp;
  if (baseElement !== undefined) {
    temp = baseElement.querySelectorAll(selector);
  } else {
    temp = document.querySelectorAll(selector);
  }
  if (temp.length === 0) {
    throw domTestingLib.queryHelpers.getElementError(
      `Unable to find an element by: selector=${selector}`,
      document.body
    );
  }
  return temp;
};

export const getSelectContentByFormLabel = (label: string) => {
  return getBySelector(
    '.ant-select-selection-item-content',
    screen.getByText(label).parentNode?.parentNode as HTMLDivElement
  );
};

export const getSelectValueByFormLabel = (label: string) => {
  return getBySelector(
    '.ant-select-selection-item',
    screen.getByLabelText(label).parentNode?.parentNode as HTMLDivElement
  );
};

export const selectOptionByIndex = (
  label: string,
  optionText: string,
  index = 1
) => {
  fireEvent.mouseDown(screen.getByLabelText(label));
  const option = screen.getAllByText(optionText)[index];
  expect(option).toHaveClass('ant-select-item-option-content');
  fireEvent.click(option);
};

/**
 * 具体功能同 selectOptionByIndex, 不过该函数主要针对下拉框选项为一个节点的情况
 * @param label 表单中 Select 组件外部 Form.Item 的label, 注意该 Form.Item 必须有 name
 * @param optionCls 下拉项节点最外层的 class
 * @param index  这里不同于 selectOptionByIndex 的 index, 其代表的就是真实需要选择的下拉项的序号, 从 0 开始.
 */
export const selectCustomOptionByClassName = (
  label: string,
  optionCls: string,
  index: number
) => {
  fireEvent.mouseDown(screen.getByLabelText(label));
  const options = getAllBySelector(`.${optionCls}`);
  let realIndex = index;
  if (index < 0) {
    realIndex = options.length + index;
  }

  const option = options[realIndex];
  expect(option.parentNode).toHaveClass('ant-select-item-option-content');
  fireEvent.click(option);
};

export const getValueByInput = (val: string) => {
  const inputList = getAllBySelector('.ant-input');
  const findInputList = [];
  for (const input of inputList) {
    if (input.getAttribute('value') === val) {
      findInputList.push(input);
    }
  }
  if (findInputList.length === 0) {
    throw domTestingLib.queryHelpers.getElementError(
      `Unable to find an element by: value=${val}`,
      document.body
    );
  }
  if (findInputList.length > 1) {
    throw domTestingLib.queryHelpers.getElementError(
      `Found multiple elements with the value=${val}`,
      document.body
    );
  }
  return findInputList[0];
};

const globalTimeout = global.setTimeout;

export const sleep = async (timeout = 0) => {
  await act(async () => {
    await new Promise((resolve) => {
      globalTimeout(resolve, timeout);
    });
  });
};

export const getHrefByText = (name: string) => {
  return screen.getByText(name).getAttribute('href');
};

export const getAllHrefByText = (name: string) => {
  return screen.getAllByText(name).map((v) => v.getAttribute('href'));
};
