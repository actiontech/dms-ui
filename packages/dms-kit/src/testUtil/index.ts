export * from './superRender';
export * from './customQuery';
export * from './common';
