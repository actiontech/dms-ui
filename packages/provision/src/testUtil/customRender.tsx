import { ThemeProvider } from '@mui/system';
import {
  render,
  renderHook,
  RenderHookOptions,
  RenderHookResult
} from '@testing-library/react';
import {
  BrowserRouter,
  MemoryRouter,
  MemoryRouterProps
} from 'react-router-dom';
import { MutableSnapshot, RecoilRoot } from 'recoil';
import SyncRecoil from '~/utils/SyncRecoil';
import { JSXElementConstructor } from 'react';
import lightTheme from '@actiontech/shared/lib/theme/light';

type RenderParams = Parameters<typeof render>;
type RecoilRootPropsCustom = {
  initializeState?: (mutableSnapshot: MutableSnapshot) => void;
  override?: boolean;
};

export const renderWithRecoilRootWithRouter = (
  ...[ui, option, recoilRootProps]: [...RenderParams, RecoilRootPropsCustom?]
) => {
  return render(
    <BrowserRouter>
      <RecoilRoot {...recoilRootProps}>{ui}</RecoilRoot>
    </BrowserRouter>,
    option
  );
};

export const renderWithRecoilRootAndRouterAndTheme = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      recoilRootProps?: RecoilRootPropsCustom;
    }?
  ]
) => {
  return render(
    <ThemeProvider theme={lightTheme}>
      <BrowserRouter>
        <RecoilRoot {...otherProps?.recoilRootProps}>{ui}</RecoilRoot>
      </BrowserRouter>
    </ThemeProvider>,
    option
  );
};

export const renderWithRecoilRootWithMemoryRouter = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      recoilRootProps?: RecoilRootPropsCustom;
    }?
  ]
) => {
  return render(
    <MemoryRouter {...otherProps?.routerProps}>
      <RecoilRoot {...otherProps?.recoilRootProps}>{ui}</RecoilRoot>
    </MemoryRouter>,
    option
  );
};

export const renderWithRecoilRootAndMemoRouterAndTheme = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      recoilRootProps?: RecoilRootPropsCustom;
    }?
  ]
) => {
  return render(
    <ThemeProvider theme={lightTheme}>
      <MemoryRouter {...otherProps?.routerProps}>
        <RecoilRoot {...otherProps?.recoilRootProps}>
          <SyncRecoil />
          {ui}
        </RecoilRoot>
      </MemoryRouter>
    </ThemeProvider>,
    option
  );
};

export const superRenderHooksWrapperFactory = (
  otherChildren?: React.ReactNode,
  recoilRootProps?: RecoilRootPropsCustom
): JSXElementConstructor<any> => {
  return ({ children }: { children: React.ReactNode }) => (
    <RecoilRoot {...recoilRootProps}>
      {otherChildren}
      {children}
      <SyncRecoil />
    </RecoilRoot>
  );
};

export const superRenderHooks = <Result, Props>(
  render: (initialProps: Props) => Result,
  options?: RenderHookOptions<Props> & {
    otherChildren?: React.ReactNode;
    recoilRootProps?: RecoilRootPropsCustom;
  }
): RenderHookResult<Result, Props> => {
  return renderHook(render, {
    wrapper: superRenderHooksWrapperFactory(
      options?.otherChildren,
      options?.recoilRootProps
    ),
    ...options
  });
};
