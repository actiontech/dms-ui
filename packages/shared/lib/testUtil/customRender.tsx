import { render, renderHook } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MutableSnapshot, RecoilRoot } from 'recoil';
import {
  BrowserRouter,
  Router,
  RouterProps,
  MemoryRouter,
  MemoryRouterProps
} from 'react-router-dom';
import { ConfigProvider, theme as antdTheme } from 'antd';
import zhCN from 'antd/locale/zh_CN';
import { Dictionary, IStore } from '../types/common.type';
import { storeFactory } from './mockRedux';
import { StyledEngineProvider, ThemeProvider } from '@mui/system';
import { mount } from 'enzyme';

import lightTheme from '../theme/light';

type MountParams = Parameters<typeof mount>;
type RenderParams = Parameters<typeof render>;
type RecoilRootPropsCustom = {
  initializeState?: (mutableSnapshot: MutableSnapshot) => void;
  override?: boolean;
};

export const renderWithRouter = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderWithServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(<Router {...props}>{ui}</Router>, option);
};

export const renderWithMemoryRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(<MemoryRouter {...props}>{ui}</MemoryRouter>, option);
};

export const renderWithRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};

export const renderWithReduxAndTheme = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(ui, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={lightTheme}>
        <Provider store={storeFactory(initStore)}>{children}</Provider>
      </ThemeProvider>
    ),
    ...option
  });
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: IStore
) => {
  return renderHook(hooks, {
    wrapper: ({ children }) => (
      <Provider store={storeFactory(storeState)}>{children}</Provider>
    )
  });
};

export const renderHooksWithReduxAndRouter = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: IStore
) => {
  return renderHook(hooks, {
    wrapper: ({ children }) => (
      <Provider store={storeFactory(storeState)}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    )
  });
};

export const renderHooksWithTheme = <TProps, TResult>(
  hooks: (props: TProps) => TResult
) => {
  return renderHook(hooks, {
    wrapper: ({ children }) => (
      <ConfigProvider
        locale={zhCN}
        theme={{ algorithm: antdTheme.defaultAlgorithm }}
      >
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
        </StyledEngineProvider>
      </ConfigProvider>
    )
  });
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(<ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>, option);
};

export const renderWithThemeAndRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(
    <MemoryRouter {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </MemoryRouter>,
    option
  );
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
      recoilRootProps?: RecoilRootPropsCustom;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <RecoilRoot {...otherProps?.recoilRootProps}>
          <Provider store={storeFactory(otherProps?.initStore)}>
            <MemoryRouter {...otherProps?.routerProps}>
              <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
            </MemoryRouter>
          </Provider>
        </RecoilRoot>
      );
    },
    ...option
  });
  return renderReturn;
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: lightTheme
    }
  });
};
