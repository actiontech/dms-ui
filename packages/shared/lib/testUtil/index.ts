export * from './superRender';
export * from './mockHook';
export * from './mockApi';
export * from './customQuery';
export * from './mockRedux';
export * from './common';
export * from './LocationDisplay';
