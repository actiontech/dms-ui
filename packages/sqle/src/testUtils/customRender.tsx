import {
  RenderResult,
  render,
  renderHook as renderHookByReact
} from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import {
  BrowserRouter,
  Router,
  RouterProps,
  MemoryRouter,
  MemoryRouterProps,
  useLocation
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { storeFactory } from './mockRedux';
import { ConfigProvider, theme as antdTheme } from 'antd';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { mount, shallow } from 'enzyme';
import zhCN from 'antd/locale/zh_CN';
import { Dictionary } from '@actiontech/shared/lib/types/common.type';

import sharedTheme from '@actiontech/shared/lib/theme/light';
import lightTheme from '../theme/light';

type RenderParams = Parameters<typeof render>;
type MountParams = Parameters<typeof mount>;
type ShallowParams = Parameters<typeof shallow>;

const themeData = {
  ...sharedTheme,
  ...lightTheme
};

export const renderWithRouter = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderWithRouterAndRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <BrowserRouter>
      <Provider store={storeFactory(initStore)}>{ui}</Provider>
    </BrowserRouter>,
    option
  );
};

export const renderWithRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: Dictionary = {}
) => {
  return renderHook(hooks, {
    wrapper: Provider,
    initialProps: {
      store: storeFactory(storeState)
    }
  } as any);
};

export const renderHooksWithTheme = <TProps, TResult>(
  hooks: (props: TProps) => TResult
) => {
  return renderHookByReact(hooks, {
    wrapper: ({ children }) => (
      <ConfigProvider
        locale={zhCN}
        theme={{ algorithm: antdTheme.defaultAlgorithm }}
      >
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
        </StyledEngineProvider>
      </ConfigProvider>
    )
  });
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
    </StyledEngineProvider>,
    option
  );
};

export const renderWithThemeAndRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </Provider>,
    option
  );
};

export const renderWithThemeAndServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(
    <Router {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </Router>,
    option
  );
};

export const renderWithThemeAndRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(
    <MemoryRouter {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </MemoryRouter>,
    option
  );
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <StyledEngineProvider injectFirst>
              <ThemeProvider theme={themeData}>{children}</ThemeProvider>
            </StyledEngineProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: themeData
    }
  });
};

export const shallowWithRouter = (...[ui, option]: [...ShallowParams]) => {
  return shallow(ui, {
    ...option,
    wrappingComponent: MemoryRouter
  });
};

export const renderLocationDisplay = (): [
  () => RenderResult,
  () => JSX.Element
] => {
  const LocationComponent = () => {
    const location = useLocation();

    return <div data-testid="location-display">{location.pathname}</div>;
  };

  return [() => renderWithRouter(<LocationComponent />), LocationComponent];
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <ThemeProvider theme={themeData}>{children}</ThemeProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};
