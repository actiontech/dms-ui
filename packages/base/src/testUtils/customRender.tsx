import { Dictionary } from '@actiontech/shared/lib/types/common.type';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/system';
import { Provider } from 'react-redux';
import { MemoryRouter, MemoryRouterProps } from 'react-router-dom';
import { storeFactory } from '@actiontech/shared/lib/testUtil/mockRedux';
import lightTheme from '@actiontech/shared/lib/theme/light';
import basePackageTheme from '../theme/light';

export type RenderParams = Parameters<typeof render>;

const themeData = {
  ...lightTheme,
  ...basePackageTheme
};

export const renderWithReduxAndTheme = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(ui, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={themeData}>
        <Provider store={storeFactory(initStore)}>{children}</Provider>
      </ThemeProvider>
    ),
    ...option
  });
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <ThemeProvider theme={themeData}>{children}</ThemeProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};
