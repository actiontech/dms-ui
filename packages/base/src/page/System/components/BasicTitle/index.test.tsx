import { renderWithTheme } from '@actiontech/shared/lib/testUtil/customRender';
import SystemBasicTitle, { SystemBasicTitleProps } from '.';


describe('base/System/components/BasicTitle', () => {
  const customRender = (params: SystemBasicTitleProps) => {
    return renderWithTheme(<SystemBasicTitle {...params} />);
  };

  it('render title', () => {
    const { baseElement } = customRender({
      title: 'title cont',
      children: <span>child node</span>
    });
    expect(baseElement).toMatchSnapshot();
  });

  it('render title when title other params', () => {
    const { baseElement } = customRender({
      title: 'title cont',
      children: 'children string',
      titleTip: 'this is title tip',
      titleExtra: <span>titleExtra</span>
    });
    expect(baseElement).toMatchSnapshot();
  });
});
