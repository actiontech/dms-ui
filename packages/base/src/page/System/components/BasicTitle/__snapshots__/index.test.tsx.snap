// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base/System/components/BasicTitle render title 1`] = `
<body>
  <div>
    <section
      class="system-form-wrapper"
    >
      <div
        class="ant-row ant-row-space-between ant-row-middle config-title-wrapper has-border"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
          >
            title cont
          </div>
        </div>
      </div>
      <span>
        child node
      </span>
    </section>
  </div>
</body>
`;

exports[`base/System/components/BasicTitle render title when title other params 1`] = `
<body>
  <div>
    <section
      class="system-form-wrapper"
    >
      <div
        class="ant-row ant-row-space-between ant-row-middle config-title-wrapper has-border"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            title cont
          </div>
          <div
            class="ant-space-item"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center basic-tooltips-wrapper css-4jbtl9"
            >
              <div
                class="ant-space-item"
              >
                <svg
                  color="#8A8F99"
                  height="14"
                  viewBox="0 0 14 14"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7 12.833A5.833 5.833 0 1 1 7 1.167a5.833 5.833 0 0 1 0 11.666m0-1.166a4.666 4.666 0 1 0 0-9.333 4.666 4.666 0 0 0 0 9.333m-.583-7.584h1.166V5.25H6.417zm0 2.334h1.166v3.5H6.417z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <span>
          titleExtra
        </span>
      </div>
      children string
    </section>
  </div>
</body>
`;
